packages/mobile/e2e/scripts/fund-e2e-accounts.ts

```diff
@@ -1,87 +1,173 @@
+            break
+          }
+        case 'cEUR':
+          try {
+            const ceurApproveTx = await ceurToken
+              .approve(ceurExchange.address, amountToExchange)
+              .send({ from: E2E_TEST_FAUCET })
+            await ceurApproveTx.waitReceipt()
+            const ceurSellAmount = await ceurExchange.quoteStableSell(amountToExchange)
+            const ceurSellTx = await ceurExchange
+              .sellStable(amountToExchange, ceurSellAmount)
+              .send({ from: E2E_TEST_FAUCET })
+            await ceurSellTx.waitReceipt()
+          } catch (err) {
+            console.log('Failed to sell cEUR', err)
+          } finally {
+            break
+          }
+      }
+    } else {
+      const toBuy = target - balances[balance]
+      const amountToExchange = kit.web3.utils.toWei(`${toBuy}`, 'ether')
+      console.log(`${balance} balance lower than ${sum / numOfTokens}`)
+      switch (balance) {
+        case 'CELO':
+          try {
+            const celoApproveTx = await celoToken
+              .approve(cusdExchange.address, amountToExchange)
+              .send({ from: E2E_TEST_FAUCET })
+            await celoApproveTx.waitReceipt()
+            const celoBuyAmount = await celoExchange.quoteGoldBuy(amountToExchange)
+            const celoBuyTx = await celoExchange
+              .buyGold(amountToExchange, celoBuyAmount)
+              .send({ from: E2E_TEST_FAUCET })
+            await celoBuyTx.waitReceipt()
+          } catch (err) {
+            console.log('Failed to buy CELO', err)
+          } finally {
+            break
+          }
+        case 'cUSD':
+          try {
+            const cusdApproveTx = await celoToken
+              .approve(cusdExchange.address, amountToExchange)
+              .send({ from: E2E_TEST_FAUCET })
+            await cusdApproveTx.waitReceipt()
+            const cusdBuyAmount = await cusdExchange.quoteStableBuy(amountToExchange)
+            const cusdBuyTx = await cusdExchange
+              .buyStable(amountToExchange, cusdBuyAmount)
+              .send({ from: E2E_TEST_FAUCET })
+            await cusdBuyTx.waitReceipt()
+          } catch (err) {
+            console.log('Failed to buy cUSD', err)
+          } finally {
+            break
+          }
+        case 'cEUR':
+          try {
+            const ceurApproveTx = await celoToken
+              .approve(ceurExchange.address, amountToExchange)
+              .send({ from: E2E_TEST_FAUCET })
+            await ceurApproveTx.waitReceipt()
+            const ceurBuyAmount = await ceurExchange.quoteStableBuy(amountToExchange)
+            const ceurBuyTx = await ceurExchange
+              .buyStable(amountToExchange, ceurBuyAmount)
+              .send({ from: E2E_TEST_FAUCET })
+            await ceurBuyTx.waitReceipt()
+          } catch (err) {
+            console.log('Failed to buy cEUR', err)
+          } finally {
+            break
+          }
+      }
+    }
+  }
```

Comment:
nit: it seems we could reduce some of the duplicated logic. No need to do it though, or maybe if we ever need to add another token.

Reply:
I could see deduplicating some of this logic as cleaner; however, I thought the ability to easily extend it to cover cREAL might be more important.